### PR TITLE
webtransport: return error before wrapping opened / accepted streams

### PR DESCRIPTION
**Pull request by marten-seemann**

### CODE REVIEW

File: p2p/transport/webtransport/conn.go [hunk 1]
```diff
@@ -46,11 +46,17 @@ func newConn(tr tpt.Transport, sess *webtransport.Session, sconn *connSecurityMu
 
 func (c *conn) OpenStream(ctx context.Context) (network.MuxedStream, error) {
 	str, err := c.session.OpenStreamSync(ctx)
+	if err != nil {
+		return nil, err
+	}
 	return &stream{str}, err
```
Comment (Jorropo) [inline p2p/transport/webtransport/conn.go:52]:
```suggestion
	return &stream{str}, nil
```

File: p2p/transport/webtransport/conn.go [hunk 2]
```diff
@@ -46,11 +46,17 @@ func newConn(tr tpt.Transport, sess *webtransport.Session, sconn *connSecurityMu
...
 }
 
 func (c *conn) AcceptStream() (network.MuxedStream, error) {
 	str, err := c.session.AcceptStream(context.Background())
+	if err != nil {
+		return nil, err
+	}
 	return &stream{str}, err
```
Comment (Jorropo) [inline p2p/transport/webtransport/conn.go:60]:
```suggestion
	return &stream{str}, nil
```

**Review by Jorropo — CHANGES_REQUESTED** [review]

**Review by Jorropo — APPROVED** [review]
Except what I sent before, SGTM

**Comment by Jorropo** [comment]
What I don't get is that there must be broken code elsewhere.
Seems like someone is ignoring err and or checking if the stream is nil.

**Comment by marten-seemann** [comment]
> What I don't get is that there must be broken code elsewhere.
> Seems like someone is ignoring err and or checking if the stream is nil instead.

That's if the bug is upstream. If the bug is downstream, we might get a `nil, nil` return, and this fix won't work (it's still the right thing to do).

**Pull request merged**